packages/ember-metal/lib/properties.ts

```diff
@@ -46,6 +50,36 @@ export abstract class Descriptor {
   didChange?(obj: object, keyName: string): void;
 }
 
+/**
+  A wrapper for a native ES5 descriptor. In an ideal world, we wouldn't need
+  this at all, however, the way we currently flatten/merge our mixins require
+  a special value to denote a descriptor.
+
+  @class NativeDescriptor
+  @private
+*/
+export class NativeDescriptor extends Descriptor {
```

Comment:
Given how this works now, I think we can move this back to its original location (in `descriptor.ts`), right?

Reply:
yep, just thought about avoiding extra import/exports :P

Reply:
done